src/lib/WalletInfoCollector.ts

```diff
@@ -25,6 +25,8 @@ type BasicAccountInfo = {
 export type WalletCollectionResult = {
     walletInfo: WalletInfo,
     receiptsError?: Error, // if there is an incomplete result due to failed requestTxReceipts requests
+    releaseKey: (removeKey: boolean) => void,
+    hasHistoryOrBalance: boolean, // if the wallet has a transaction history or a balance
```

Comment:
Should this rather be called `hasActivity` (or `wasUsed` /  `hasUsage`) to match the `checkLegacyActivity`?
```suggestion
    hasActivity: boolean, // whether the wallet has a transaction history or a balance or owns a contract
```